algorithms/bellman_ford.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+import graphviz
+
+## Defining classes for vertex and Graph
+
+class vertex :
+    def __init__(self,distance=0,pre=0):
+        self.d=distance
+        self.p=pre
+
+        
+class Graph :
+    def __init__(self,vertices=0,edges=0):
+
+        self.vertices=vertices
+        self.edges=edges
+   
+    def single_source(self,j):
+        k=list(self.edges.values())
+        k=np.array(k)
+        k=abs(k)
+        max_edge= k.sum()
+        for i in self.vertices:
+            self.vertices[i].d= max_edge+1
+        self.vertices[j].d=0
+        return None
+            
+    def relax(self,j):
+        if self.vertices[int(j[3])].d > self.vertices[int(j[1])].d +self.edges[j]:
+           self.vertices[int(j[3])].d =  self.vertices[int(j[1])].d +self.edges[j]
+           self.vertices[int(j[3])].p= int(j[1])
+           return None
+             
+        
+
```

Comment:
Usual convention is to leave two lines before function/class definitions. Follow this in the whole file.

**TIP**: Following code convention is very important in a big project as multiple people may have different styles of writing code. Hence, it is important for everyone have consistent changes.